src/host/outputStream.cpp

```diff
@@ -770,3 +770,30 @@ bool ConhostInternalGetSet::PrivateIsVtInputEnabled() const
 {
     return _io.GetActiveInputBuffer()->IsInVirtualTerminalInputMode();
 }
+
+// Method Description:
+// - Updates the buffer's current text attributes depending on whether we are
+//   starting/ending a hyperlink
+// Arguments:
+// - The hyperlink URI
+// Return Value:
+// - true
+bool ConhostInternalGetSet::PrivateAddHyperlink(const std::wstring_view uri, const std::wstring_view params) const
+{
+    auto attr = _io.GetActiveOutputBuffer().GetAttributes();
+    if (uri.empty())
+    {
+        // URI is empty, this means we are ending a hyperlink
+        attr.SetHyperlinkId(0);
+        _io.GetActiveOutputBuffer().GetTextBuffer().SetCurrentAttributes(attr);
+    }
+    else
+    {
+        // URI is non-empty, this means we are starting a hyperlink
+        USHORT id = _io.GetActiveOutputBuffer().GetTextBuffer().GetHyperlinkId(params);
+        attr.SetHyperlinkId(id);
+        _io.GetActiveOutputBuffer().GetTextBuffer().SetCurrentAttributes(attr);
+        _io.GetActiveOutputBuffer().GetTextBuffer().AddHyperlinkToMap(uri, id);
+    }
+    return true;

```

Comment:
This is the nit-iest of nits, but typically we have another function in `getset.cpp` that actually handles the implementation of these functions, something like `DoSrvPrivateAddHyperlink`. This file's pretty much just full of wrappers around those methods. (This made more sense when more of the VT api was implemented using the Console API, but since so much of the VT API needed it's own functions(the Private* ones), this file has become full of basically just wrappers)